### PR TITLE
Generate GHEC

### PR DESCRIPTION
**Pull request by kfcampbell**
There's still a lot of duplication right now between the staged directories, which I don't like and I'd love to find a way out of. Maybe that's inevitable and keeping it as close together as possible like we're doing now is the best we can do.

This PR adds the machinery to generate GHEC SDKs to https://github.com/octokit/go-sdk-enterprise-cloud and https://github.com/octokit/dotnet-sdk-enterprise-cloud. The actual content of those SDKs are stored in the stage directory of this repository and have not yet been PR'd to the downstream repositories or released.

### CODE REVIEW

File: schemas/main.go
Comment (kfcampbell) [inline schemas/main.go:64]: This error was removed to simplify logic in the `scripts/generate-{whatever}.sh` script: there, we can specify the version every time through an environment variable, and then simply ignore it here if it's blank.

File: scripts/generate-dotnet.sh
Comment (kfcampbell) [inline scripts/generate-dotnet.sh:6]: It may be smart to refactor this type of logic into some kind of tools script, and then import functions between scripts. Perhaps that's a job for a future PR?

File: scripts/generate-dotnet.sh
Comment (kfcampbell) [inline scripts/generate-dotnet.sh:23]: This may be overkill; another option here would be removing version checking altogether and then erroring out in the schema download if the version doesn't exist in the `github/rest-api-description` repo.

File: scripts/generate-go.sh
Comment (kfcampbell) [inline scripts/generate-go.sh:22]: See above comment about version checking.

File: stage/dotnet/dotnet-sdk-enterprise-cloud/GitHub.Octokit.sln
Comment (kfcampbell) [inline stage/dotnet/dotnet-sdk-enterprise-cloud/GitHub.Octokit.sln:15]: I kept the solution name the same here and changed the .csproj to `GitHub.Octokit.GHEC.SDK.csproj`: note the addition of `GHEC` in the middle there. Thoughts?

File: .github/workflows/build-dotnet.yml
Comment (nickfloyd) [inline .github/workflows/build-dotnet.yml:11]: Just curious, why did you use the moniker `cloud-builds` here? I'm guessing it has to do with the hosted environment and you wanted to index / group on that.

File: stage/dotnet/dotnet-sdk-enterprise-cloud/GitHub.Octokit.sln
Comment (kfcampbell) [reply stage/dotnet/dotnet-sdk-enterprise-cloud/GitHub.Octokit.sln:15]: What you're describing was my reasoning exactly. I'm picturing for GHES it looks something like `GitHub.Octokit.GHES.3.19.SDK`.

File: scripts/generate-dotnet.sh
Comment (nickfloyd) [reply scripts/generate-dotnet.sh:6]: I would hold off on abstraction until all of the dust settles from all of the changes for the enterprise effort... my guess is that we'll find other areas of optimization as well.

File: scripts/generate-dotnet.sh
Comment (nickfloyd) [reply scripts/generate-dotnet.sh:23]: Possibly.  Doing so might help us avoid the whole ✅ on double digit version number parts. But I am pretty sure the expression above will catch that.  Were you thinking if the file fetch ends up in a 404 then log and move on?

File: stage/dotnet/dotnet-sdk-enterprise-cloud/README.md
Comment (nickfloyd) [inline stage/dotnet/dotnet-sdk-enterprise-cloud/README.md:20]: My guess is that this snipit might need to change if we decide to namespace differently in the event that users would want to implement dotcom and ghe* sdks in one app - like one where they can choose their platform and authenticate. Though I suppose they could alias.  I'm just trying to picture the use case, though it might not be an actual use case.  Either way these examples might need to be updated.

File: stage/dotnet/dotnet-sdk-enterprise-cloud/README.md
Comment (nickfloyd) [inline stage/dotnet/dotnet-sdk-enterprise-cloud/README.md:3]: We should probably add a note about the need to define a host in the headers and with the SDK instantiation itself.

Comment (nickfloyd) [reply stage/dotnet/dotnet-sdk-enterprise-cloud/README.md:3]: Actuallly I think we talked about modifying the middleware to do that for the user.  Does that sound right?

File: scripts/generate-dotnet.sh
Comment (kfcampbell) [reply scripts/generate-dotnet.sh:23]: Yep, exactly. That would happen [here](https://github.com/octokit/source-generator/blob/main/schemas/main.go#L94-L98) without any additional effort on our part.

File: .github/workflows/build-dotnet.yml
Comment (kfcampbell) [reply .github/workflows/build-dotnet.yml:11]: Uhh...you know, it seems like pretty poor naming. I guess I did it because right now it only comprises of cloud-based products, but I'm not sure how that'll fit when we work in GHES.

Comment (kfcampbell) [reply .github/workflows/build-dotnet.yml:11]: Maybe `sdk-builds`? `builds` would probably be fine too; I think I changed it because I originally got confused and thought it might have been a reserved Actions keyword.

File: stage/dotnet/dotnet-sdk-enterprise-cloud/README.md
Comment (kfcampbell) [reply stage/dotnet/dotnet-sdk-enterprise-cloud/README.md:3]: I think this should be done automatically for the user by default, and then they have the option of changing it if they'd like. So perhaps both options are appropriate, and that might be best done in a PR to the dotnet-sdk-enterprise-cloud repository after the initial PR is merged. 

There will be some additional cleanup required there, I'm sure.

Comment (kfcampbell) [reply stage/dotnet/dotnet-sdk-enterprise-cloud/README.md:3]: Right now the project builds and I've made a best-effort pass at correcting dotnet-sdk references, but those build badges in the above links for example still reference dotnet-sdk, which need to be changed.

File: stage/dotnet/dotnet-sdk-enterprise-cloud/README.md
Comment (kfcampbell) [reply stage/dotnet/dotnet-sdk-enterprise-cloud/README.md:20]: Agreed! I don't believe we've made a decision on aliasing or namespacing yet; I don't have a strong preference either way. Do you?

**Review by nickfloyd — APPROVED** [review]

**Pull request merged**
